.github/workflows/pr_and_main.yml

```diff
@@ -0,0 +1,86 @@
+name: PR and main CI
+
+on:
+  push:
+    branches:
+      - main
+  pull_request:
+
+concurrency:
+  group: ${{ github.head_ref }}
+  cancel-in-progress: true  # cancel jobs from previous push
+
+jobs:
+  formatting:
+    runs-on: ubuntu-latest
+
+    steps:
+      - uses: actions/checkout@v2
+      - run: sudo apt-get install --yes clang-format-10
+      - run: find lib -type f -regex '.*\.\(cpp\|h\|tcc\)' -exec clang-format-10 -i {} +
+      - run: pip install cmake_format==0.6.9 flake8 yapf==0.30.0
```

Comment:
This duplicates the version info from the conda env files. Is there a way to use those files here?

Reply:
Only if we want to use a conda env here, which we may not for performance reasons? Note that we have the same duplication in the current pipelines setup.

Reply:
Could be solved with a template repo

Reply:
Not sure what you have in mind.

Reply:
If we generate this file together with the env files from templates, we can specify the versions once and paste them into all files that need them.

Reply:
👍